src/ldp.rs

```diff
@@ -1072,6 +1111,199 @@ impl ProofSuite for Eip712Signature2021 {
     }
 }
 
+async fn micheline_from_document_and_options(
+    document: &(dyn LinkedDataDocument + Sync),
+    proof: &Proof,
+) -> Result<Vec<u8>, Error> {
+    let doc_dataset = document.to_dataset_for_signing(None).await?;
+    let doc_dataset_normalized = urdna2015::normalize(&doc_dataset)?;
+    let doc_normalized = doc_dataset_normalized.to_nquads()?;
+    let sigopts_dataset = proof.to_dataset_for_signing(Some(document)).await?;
+    let sigopts_dataset_normalized = urdna2015::normalize(&sigopts_dataset)?;
+    let sigopts_normalized = sigopts_dataset_normalized.to_nquads()?;
+    let string_bytes = [
+        "Tezos Signed Message: ".to_string(),
+        sigopts_normalized,
+        doc_normalized,
+    ]
+    .join("\n")
+    .as_bytes()
+    .to_vec();
+    const BYTES_PREFIX: [u8; 2] = [05, 01];
+    let data = [
+        BYTES_PREFIX.to_vec(),
+        string_bytes.len().to_be_bytes().to_vec(),
+        string_bytes,
+    ]
+    .concat();
+    Ok(data)
+}
+
+pub struct TezosSignature2021;
```

Comment:
I was wondering, I'm using that directly but do we want to make these private to force the end-user to use `LinkedDataProofs` and simplify the API (although it could also make it more opaque but it's another problem).

Reply:
I think we want to move towards having these proof suites be removed from ssi into separate crates, like how the DID method implementations were (https://github.com/spruceid/ssi/pull/67): https://github.com/spruceid/ssi/issues/174
So it will need to be public eventually, but in another crate. The interfaces may need to change as part of that too.

The tests in `did-pkh` rely on the proof suite type being public:
https://github.com/spruceid/ssi/pull/170/files#diff-fe6bb6d254a5437997ba7be9e87482cbabc6ce8e242c5970302db8fc1e68e5e1R886
Also mentioned here: https://github.com/spruceid/ssi/pull/124#discussion_r599007293.
This is used to create proofs when `LinkedDataProofs::sign` would not do it, for testing - in particular negative testing such as the following:
https://github.com/spruceid/ssi/pull/170/files#diff-fe6bb6d254a5437997ba7be9e87482cbabc6ce8e242c5970302db8fc1e68e5e1R669-R677